src/Modules/Server.cpp

```diff
@@ -708,6 +709,17 @@ float hostTimeWrap() {
 	return engine->GetHostTime();
 }
 
+// UTIL_EntityByIndex
+extern Hook g_EntityByIndexHook;
+DETOUR_T(int, Server::EntityByIndex, int entityIndex) {
+	if (entityIndex > Offsets::NUM_ENT_ENTRIES) return NULL;
```

Comment:
I guess this is supposed to be the real bugfix that "Valve" missed?

Also `NULL` should only be used for pointers:
```suggestion
	if (entityIndex > Offsets::NUM_ENT_ENTRIES) return 0;
```

Reply:
The function we're hooking uses NULL but sure

Reply:
ok so `UTIL_EntityByIndex` returns type `CBaseEntity *`